<?xml version="1.0" ?>
<launch>
    <machine name="moma-dell-workstation" address="moma-dell-workstation" env-loader="/home/spadmin/perception/env.sh" user="spadmin" default="never"/>

    <arg name="visualize" default="false" />

    <node machine="moma-dell-workstation" name="mask_rcnn" pkg="mask_rcnn_ros" type="mask_rcnn_node.py" output="log">
        <remap from="~input" to="/camera/color/image_raw" />
        <param name="~visualization" value="$(arg visualize)" />
    </node>

    <arg name="depth_segmentation_node_params_name" default="/home/franka/catkin_ws/src/moma/moma_demos/grasp_demo/config/voxbloxpp_cfg/franka_config.yaml"/>

    <node machine="moma-dell-workstation" name="depth_segmentation_node" pkg="depth_segmentation" type="depth_segmentation_node" output="screen">
        <rosparam command="load" file="$(arg depth_segmentation_node_params_name)" />
        <param name="semantic_instance_segmentation/enable" value="true"/>
        <param name="label_display" value="$(arg visualize)"/>
    </node>

    <arg name="gsm_node_params_name" default="/home/franka/catkin_ws/src/moma/moma_demos/grasp_demo/config/voxbloxpp_cfg/franka.yaml"/>

    <node machine="moma-dell-workstation" name="gsm_node" pkg="gsm_node" type="gsm_node" output="screen">
        <rosparam command="load" file="$(arg gsm_node_params_name)" />
        <param name="meshing/visualize" value="$(arg visualize)"/>
    </node>

</launch>
